### ISSUE TITLE
cantaloupe sometimes answers a valid request with HTTP error 403

### ISSUE THREAD

**Issue by sebschlue** [cantaloupe-project/cantaloupe#422]
I am using cantaloupe 4.1.5 in a production setup and notice a certain amount of 403 errors for which I don't know the reason. It seems that the percentage of these errors increases when the load increases.

Here is my very basic config, simple HTTP source and Azure Cache:

```
endpoint.api.enabled = true
endpoint.api.username = api
endpoint.api.secret = xxxxxxxxxxxxxxxxxxxx
source.static = HttpSource
HttpSource.request_timeout = 30
HttpSource.BasicLookupStrategy.url_prefix = https://www.<<host>>.<<domain>>/images/
cache.server.source = 
cache.server.source.ttl_seconds = 86400
cache.server.derivative.enabled = true
cache.server.derivative = AzureStorageCache
cache.server.derivative.ttl_seconds = 86400
cache.server.worker.enabled = true
cache.server.worker.interval = 3600
AzureStorageCache.account_name = xxxxxxxxxxxxxxxxxxxx
AzureStorageCache.account_key = xxxxxxxxxxxxxxxxxxxx
AzureStorageCache.container_name = xxxxxxxxxxxxxxxxxxxx
AzureStorageCache.object_key_prefix = xxxxxxxxxxxxxxxxxxxx
log.application.level = warn
```

There are only two kinds of log messages:

The first:

`[work-nm-5666738] [1;31mERROR[0;39m [36me.i.l.c.c.InfoService[0;39m - putInCachesAsync(): null`

The second:

`[qtp300031246-139927] [31mWARN [0;39m [36me.i.l.c.s.HealthChecker[0;39m - Health check completed in 2 msec: RED: java.util.concurrent.ExecutionException: java.io.EOFException: HttpConnectionOverHTTP@3269cf3e::DecryptedEndPoint@6ba360bf{www.<<host>>.<<domain>>/<<ip>>:443<->/10.42.2.64:58920,OPEN,fill=-,flush=-,to=733/0} (<<image>>.jpg -> edu.illinois.library.cantaloupe.source.HttpSource -> edu.illinois.library.cantaloupe.processor.Java2dProcessor)`

But the timestamps of these logs do NOT correspond to the occurrences of HTTP 403 errors.

Can you maybe tell me more about these two log entries and also give me a hint as to which circumstances may lead to 403 errors? 

Thanks in advance :)

**Comment by adolski** [comment]
Hi @sebschlue,

Try changing the value of your `source.static` config key to `HttpSource2`. This is a different implementation of HttpSource backed by an OkHttp client instead of Jetty. The Jetty client is buggy and has caused problems for some users. The second log message you posted looks suspiciously like something it would do when it's acting up.

In version 5.0, HttpSource2 replaces the current HttpSource, which has been removed, and takes its name. There will also be a `log_error_responses` configuration key which may help in diagnosing situations like this where the application is returning an error status with no apparent corresponding error in the logs.

**Comment by sebschlue** [comment]
Hi @adolski ,

thank you very much for your reply. I have tried this setting an it does not seem to work for me. Cantanloupe did not serve images and almost immediately crashed with Java OOM.

```
2021-01-11 09:57:54.836:INFO::main: Logging initialized @344ms to org.eclipse.jetty.util.log.StdErrLog
2021-01-11 09:57:54.932:INFO:oejs.Server:main: jetty-9.4.24.v20191120; built: 2019-11-20T21:37:49.771Z; git: 363d5f2df3a8a28de40604320230664b9c793c16; jvm 11.0.7+10
2021-01-11 09:57:56.580:INFO:oejw.StandardDescriptorProcessor:main: NO JSP Support for /, did not find org.eclipse.jetty.jsp.JettyJspServlet
2021-01-11 09:57:57.186:INFO:oejs.session:main: DefaultSessionIdManager workerName=node0
2021-01-11 09:57:57.186:INFO:oejs.session:main: No SessionScavenger set, using defaults
2021-01-11 09:57:57.189:INFO:oejs.session:main: node0 Scavenging every 600000ms
Loading config file: /config/cantaloupe.properties
09:57:57.317 [main] WARN  e.i.l.c.ApplicationContextListener - The effective temp directory is the system temp directory. Severe errors may result if the application is left running for a long time. Consider setting `temp_pathname` for long-running servers.
2021-01-11 09:57:57.642:INFO:oejsh.ContextHandler:main: Started o.e.j.w.WebAppContext@189b5fb1{Cantaloupe,/,file:///tmp/jetty-0_0_0_0-8182-cantaloupe-4_1_5_war-_-any-3855752028202759907.dir/webapp/,AVAILABLE}{/opt/cantaloupe/cantaloupe-4.1.5.war}
2021-01-11 09:57:57.665:INFO:oejs.AbstractConnector:main: Started ServerConnector@52e677af{HTTP/1.1,[http/1.1]}{0.0.0.0:8182}
2021-01-11 09:57:57.666:INFO:oejs.Server:main: Started @3199ms
Terminating due to java.lang.OutOfMemoryError: Java heap space
```

Test was run with JDK_JAVA_OPTIONS == '-Xmx2g -XX:+ExitOnOutOfMemoryError'

**Comment by adolski** [comment]
@sebschlue That OutOfMemoryError is not related to this original issue. You would need to consider the request URL and format/size characteristics of the source image to understand why it's happening. Increasing the log level to `info` or `debug` could also provide more information.
